api_app/script_analyzers/observable_analyzers/urlscan.py

```diff
@@ -0,0 +1,81 @@
+import requests
+import time
+import json
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class UrlScan(classes.ObservableAnalyzer):
+    base_url: str = "https://urlscan.io/"
+
+    def set_config(self, additional_config_params):
+        self.analysis_type = additional_config_params.get("urlscan_analysis", "search")
+        api_key_name = additional_config_params.get("api_key_name", "URLSCAN_KEY")
+        self.visibility = additional_config_params.get("visibility", "public")
+        self.__api_key = secrets.get_secret(api_key_name)
+
+    def __submit_query(self):
+        data = {"url": self.observable_name, "visibility": self.visibility}
+        uri = "api/v1/scan/"
+        try:
+            response = requests.post(
+                self.base_url + uri, headers=self.headers, data=json.dumps(data)
+            )
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+        return response.json()["api"]
+
+    def __poll_for_result(self, url, headers):
+        max_tries: int = 10
+        poll_distance: int = 2
+        result = {}
+        for chance in range(max_tries):
+            time.sleep(poll_distance)
+            resp = requests.get(url, headers=headers)
+            resp_json = resp.json()
+            status = resp_json.get("status", None)
+            if status == "done":
```

Comment:
as stated in URLScan docs, I think you should make this check based on the status code and not from the keyword "status"

`The response to the API call will give you the scan ID and API endpoint for the scan, you can use it to retrieve the result after waiting for a short while. Until the scan is finished, the URL will respond with a HTTP 404 status code.`

Reply:
Changed poll_for_result. It follows the procedure mentioned in URLScan docs.